pkg/pipelines/webhook/webhook.go

```diff
@@ -106,10 +110,37 @@ func newWebhookInfo(accessToken, pipelinesFile string, serviceName *QualifiedSer
 	if err != nil {
 		return nil, fmt.Errorf("failed to get event listener URL: %v", err)
 	}
-
+	if accessToken == "" {
+		accessToken, err = getAccessToken(gitRepoURL)
+		if err != nil {
+			return nil, err
+		}
+	}
 	return &webhookInfo{clusterResources, repository, gitRepoURL, cicdNamepace, listenerURL, accessToken, serviceName, isCICD}, nil
 }
 
+func getAccessToken(gitRepoURL string) (string, error) {
+	accessToken, _ := keyring.Get("kam", gitRepoURL)
+	if accessToken == "" {
+		parsed, err := url.Parse(gitRepoURL)
+		if err != nil {
+			return "", fmt.Errorf("failed to parse repository URL %q: %w", gitRepoURL, err)
+		}
+		repoNameExt, err := git.GetRepoName(parsed)
+		if err != nil {
+			return "", err
+		}
+		repoName := strings.Split(repoNameExt, "/")[1]
+		envVar := strings.ToLower(repoName) + "GitToken"
```

Comment:
Usually env-vars are are all in upper-case, rather than mixed or lower-case.

https://google.github.io/styleguide/shellguide.html#constants-and-environment-variable-names